diff --git a/node_modules/minecraft-protocol/src/client.js b/node_modules/minecraft-protocol/src/client.js
index e369e77..7d192bf 100644
--- a/node_modules/minecraft-protocol/src/client.js
+++ b/node_modules/minecraft-protocol/src/client.js
@@ -77,7 +77,7 @@ class Client extends EventEmitter {
       e.field = [this.protocolState, deserializerDirection].concat(parts).join('.')
       e.message = e.buffer ? `Parse error for ${e.field} (${e.buffer?.length} bytes, ${e.buffer?.toString('hex').slice(0, 6)}...) : ${e.message}` : `Parse error for ${e.field}: ${e.message}`
       if (!this.compressor) { this.splitter.pipe(this.deserializer) } else { this.decompressor.pipe(this.deserializer) }
-      this.emit('error', e)
+      if (e.name !== 'PartialReadError' && !e.message?.includes('array size is abnormally large')) this.emit('error', e)
     })
     this._mcBundle = []
     const emitPacket = (parsed) => {
diff --git a/node_modules/minecraft-protocol/src/client/play.js b/node_modules/minecraft-protocol/src/client/play.js
index 71ad739..9d4c693 100644
--- a/node_modules/minecraft-protocol/src/client/play.js
+++ b/node_modules/minecraft-protocol/src/client/play.js
@@ -53,14 +53,17 @@ module.exports = function (client, options) {
         client.write('configuration_acknowledged', {})
       }
       client.state = states.CONFIGURATION
+      client.removeAllListeners('select_known_packs')
       client.once('select_known_packs', () => {
         client.write('select_known_packs', { packs: [] })
       })
+      client.removeAllListeners('code_of_conduct')
       client.once('code_of_conduct', () => {
         client.write('accept_code_of_conduct', {})
       })
       // Server should send finish_configuration on its own right after sending the client a dimension codec
       // for login (that has data about world height, world gen, etc) after getting a login success from client
+      client.removeAllListeners('finish_configuration')
       client.once('finish_configuration', () => {
         client.write('finish_configuration', {})
         client.state = states.PLAY
